extract_endpoint/src/extract_endpoint/azure_utils.py

```diff
@@ -17,16 +18,66 @@ class EnvVariables(enum.Enum):
     domain = 'EXTRACT_ENDPOINT_STORAGE_DOMAIN'
 
 
-def upload_bytes_to_azure(coords: StorageCoordinates, data: bytes, filename: str) -> bool:
-    azure_service = blob.BlockBlobService(account_name=coords.account,
-                                          account_key=coords.key,
-                                          custom_domain=coords.domain)
-    azure_service.create_blob_from_bytes(coords.container, filename, data)
-    return filename in [blob.name for blob in azure_service.list_blobs(coords.container)]
+class StorageProtocol(typing_extensions.Protocol):
+    def upload(self, data: bytes, filename: str) -> bool:
+        pass
 
+    def download(self, filename: str) -> str:
+        pass
 
-def download_bytes_from_azure(coords: StorageCoordinates, filename: str) -> str:
-    azure_service = blob.BlockBlobService(account_name=coords.account,
-                                          account_key=coords.key,
-                                          custom_domain=coords.domain)
-    return azure_service.get_blob_to_bytes(coords.container, filename).content
+
+class LocalStorage:
+    def __init__(self) -> None:
+        self._upload_run_count = 0
+        self._download_run_count = 0
+
+    @property
+    def upload_run_count(self) -> int:
+        return self._upload_run_count
+
+    @property
+    def download_run_count(self) -> int:
+        return self._download_run_count
+
+    def upload(self, data: bytes, filename: str) -> bool:
```

Comment:
you don't seem to be reading or writing from/to disk...

Reply:
LocalStorage should probably be renamed TestStorage to be more specific. The goal of it isn't to store or download anything, rather to allow us to test in our unit tests whether the function we expect to be called is & that it's called the right number of times